### ISSUE TITLE
So far so good but...

### ISSUE THREAD

**Issue by ssamayoa** [NEORIS-ZDEVOPS/BMS-Designer#5]
I had to create new BMS for a small project and really appreciate the existence of this plugin, however:

I wish it haves:
- Group select
- Change properties for selected items
- Copy/paste
- Language server? 
- Integration with Zowe Explorer / Z Open Editor?
- Moving items with the keyboard.

Small bugs:
- Drop down are shown elsewhere in the screen.
- Clearing a property does not removes it.
- Changing size does not changes INIT in input field.
 
BOTTOM LINE:
Good enough for small projects but I wish is open source.  

THANKS!

**Comment by antonkc** [comment]
Thanks for the review, @ssamayoa. We will take it into account for the following update to the extension.

Regarding making the project open source. I undestand your wish, I also see that this is your second time asking. But we can't provide, at least, in the short term. If the internal situation turns out favorable for such thing, I will be proposing it to the team.

**Comment by ssamayoa** [comment]
@antonkc 
Any news on this plug-in?
Haven't been updated since Aug 2024 :(

**Comment by antonkc** [comment]
Greetings @ssamayoa ,

Sorry for the delay. We don't have enough people currently to attend to all projects, internal and public, as much as ideal.

I will try to forward this to the team.

**Comment by phaumer** [comment]
@antonkc you published your extension in the VS Code Marketplace with an open-source Apache license, which gives everyone the right to copy, modify, and redistribute your extension. However, this repo does not have the source code. Providing that here in GitHub would make it easier for the community to contribute and to extend your code.

**Comment by ssamayoa** [comment]
> [@antonkc](https://github.com/antonkc) you published your extension in the VS Code Marketplace with an open-source Apache license, which gives everyone the right to copy, modify, and redistribute your extension. However, this repo does not have the source code. Providing that here in GitHub would make it easier for the community to contribute and to extend your code.

I guess they want to monetize somehow / sometime in the future and/or have already paying customers so making it OS would force them to stop charging to those customers - who knows!

I asked for the code almost a year ago but it seems that they aren't willing to release it.

**Comment by antonkc** [comment]
Welcome @phaumer,

Z-Open Editor's repository also has the same license and you do not share the source code. My guess is that you are just trying to improve the current vscode environment? That's actually fair. Anyway, I did relay your message. No promises, but my boss said that we will be studying with legal how to publish the source code.

@ssamayoa, we are not planning to monetize this extension. It is just that all human resources that did contribute to the extension are with their hands full with a different project (and, yes, it is sadly because of money).

Kind regards.

**Comment by phaumer** [comment]
@antonkc Z Open Editor is not Apache licensed. When you click the License link in the marketplace you see our commercial IBM non-warranty license(<https://marketplace.visualstudio.com/items/IBM.zopeneditor/license>), i.e. you can use it for free without warranty, but you cannot disassemble, modify, or redistribute. 

We have an Apache license in our GitHub repo for issues, which indicates that all the code samples that people post on the site as part of an issue they report will automatically be Apache licensed so we are allowed to use the submitted code to reproduce problems. We explain it in our readme: <https://github.com/IBM/zopeneditor-about?tab=readme-ov-file#license>

Your extension looks really promising. If it becomes open-source some day we would have a look.

**Comment by ssamayoa** [comment]
@phaumer 
So you work for IBM?
Would be nice if NEORIS-ZDEVOPS donated the code then IBM keep them OS and let other people to improve it.

**Comment by phaumer** [comment]
It would depend on the license. Donating it to Zowe (Zowe Explorer for CICS) could be another path, but it would require a core group of developers to maintain it.

**Comment by antonkc** [comment]
About the license, It is meant to be more proprietary than the license to this repo. The license to this repo applies only to the contents that are published. If that makes sense. Honestly, I just want to develop and don't think too much about licenses. But with my current assignations don't allow me to spend a lot of time with the extensions.

I can promise something though. Each time I see a comment here I will forward it to people in Neoris. And by here I mean, in the repository. I would appreciate if any of you opened an issue about the license/transfer. This is not the issue for that.

Best regards.
